### DISCUSSION TITLE
Requesting self-description

### DISCUSSION THREAD

**Discussion by MikhailGordienk** [International-Data-Spaces-Association/DataspaceConnector#75] [Q&A] [answered]
Hello!

The documentation is a bit confusing. There is no public **/api/ids/data** defined in the Controller-level. Instead, there is **/admin/api/request/description** endpoint which is protected.  How a user is supposed to provide credentials to access another connectors' self-description? Considering two connectors demo-scenario I see no option but to disable security on both sides.

The same is for certificates. If I want to run two connectors and make one of them consume resources from another I have to generate a new certificate and share it between them.

**Comment by juliapampus** [comment, marked as answer]
The `/api/ids/data endpoint` is provided by the framework dependency. That is why you cannot find an implementation in the DSC code. To request a connector's self-description, send a request via `POST https://{consumer}/admin/api/request/description` to `POST https://{provider}/api/ids/data` as recipient.

**Reply by thilo20** [reply]
I just tested this successfully with the current release DSC-v5.2.0 using the interactive Swagger UI:
consumer=https://localhost:8081
provider=https://localhost:8080

Nagivate to https://localhost:8081/api/swagger-ui/index.html?configUrl=/v3/api-docs/swagger-config#/IDS%20Messages/sendDescriptionRequestMessage
- recipient=https://localhost:8080/api/ids/data
- elementId='leave empty'

Result is 200, response body is provider connector self description, contains..
 "ids:resourceCatalog" : [ {
    "@type" : "ids:ResourceCatalog",
    "@id" : "https://localhost:8080/api/catalogs/73666d86-19e2-465b-8b54-da0455566cb4"

From here you can navigate the catalog, offers etc. by giving the @id value as 'elementId' parameter and hitting 'Execute' again ✌️
